### PR TITLE
Maulik/mfe preview changes

### PR DESCRIPTION
**Pull request by Maulik003**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by Maulik003** [comment]
[https://github.com/Azure/azure-openapi-validator/issues/248](https://github.com/Azure/azure-openapi-validator/issues/248)
[https://github.com/Azure/azure-rest-api-specs/issues/20235](https://github.com/Azure/azure-rest-api-specs/issues/20235)

References of issues for above required failing checks

**Comment by raosuhas** [comment]
@Maulik003 Is the first checkin the previous version ? I cant make it out since you have updated the version field. If not then please follow the process explained here : 
image

**Comment by forteddyt** [comment]
> @Maulik003 Is the first checkin the previous version ? I cant make it out since you have updated the version field. If not then please follow the process explained here : image

@raosuhas the first commit is the changes we're making for 2022-10-01-preview. The 2022-10-01-preview API is following the staging branch paradigm (note that we're merging into a staging branch, not main). This PR is adding the changes for the mfe.json file only

I'm confused on what you mean by your comment

**Comment by raosuhas** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/registries/{registryName}/codes/{name}": {

please update to codename rather than just name

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/mfe.json:67 in c4d31c6. [](commit_id = c4d31c6578d3f002bd7664029c130748b4eff063, deletion_comment = False)

**Comment by raosuhas** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/registries/{registryName}/components/{name}": {

componentname

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/mfe.json:652 in c4d31c6. [](commit_id = c4d31c6578d3f002bd7664029c130748b4eff063, deletion_comment = False)

**Comment by raosuhas** [comment]
>             "name": "listViewType",

can you use ODATA $filter to do this ?

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/mfe.json:1214 in c4d31c6. [](commit_id = c4d31c6578d3f002bd7664029c130748b4eff063, deletion_comment = False)

**Comment by raosuhas** [comment]
>                 {

again can you use ODATA filter ? looks like you are already using it for top and orderby

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/mfe.json:1549 in c4d31c6. [](commit_id = c4d31c6578d3f002bd7664029c130748b4eff063, deletion_comment = False)

**Comment by raosuhas** [comment]
>             ],

some comment everywhere in this PR for filter

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/mfe.json:1861 in c4d31c6. [](commit_id = c4d31c6578d3f002bd7664029c130748b4eff063, deletion_comment = False)

**Comment by raosuhas** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/registries/{registryName}/environments/{name}": {

enviromentname

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/mfe.json:1264 in c4d31c6. [](commit_id = c4d31c6578d3f002bd7664029c130748b4eff063, deletion_comment = False)

**Comment by raosuhas** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/registries/{registryName}/environments/{name}": {

same comment everywhere in this PR for name

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/mfe.json:1264 in c4d31c6. [](commit_id = c4d31c6578d3f002bd7664029c130748b4eff063, deletion_comment = False)

**Comment by raosuhas** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/registries/{registryName}/models/{name}/versions/{version}": {

It seems to be the case , but can you please confirm if all these new resources you are adding are Proxy Resources ?

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/mfe.json:2246 in c4d31c6. [](commit_id = c4d31c6578d3f002bd7664029c130748b4eff063, deletion_comment = False)

**Comment by mentat9** [comment]
Still a couple of @raosuhas comments you haven't responded to ($filter recommendation and question about proxy resources).

**Comment by Maulik003** [comment]
> Still a couple of @raosuhas comments you haven't responded to ($filter recommendation and question about proxy resources).

All new resources that are added are proxy resources.
Regarding $filter, teams thread is going on to discuss more on that with Suhas, Teddy and Wenbin

**Comment by mentat9** [comment]
> Regarding $filter, teams thread is going on to discuss more on that with Suhas, Teddy and Wenbin

Please leave the ARMChangesRequested label until that thread reaches a conclusion.

**Comment by rkmanda** [comment]
>             "name": "listViewType",

The non-OData style query params like listViewType were originally introduced in GA API version and we need to allow new resource types to follow a similar pattern for the sake of maintaining consistency within the context of the namespace. 

---
In reply to: [1235859712](https://github.com/Azure/azure-rest-api-specs/pull/20372#issuecomment-1235859712) [](http://example.com/codeflow?ancestors=1235859712,1235859712)

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/mfe.json:1214 in c4d31c6. [](commit_id = c4d31c6578d3f002bd7664029c130748b4eff063, deletion_comment = False)

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/mfe.json [hunk 1]
```diff
@@ -9,6 +9,2478 @@
     "https"
   ],
   "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/registries/{registryName}/codes": {
```
Comment (ArcturusZhang) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/mfe.json:12]: This swagger file does not have a globally defined "consumes" and "produces" attributes, nor do we have in these new operations, therefore this is causing our SDK generations to fail.
Could you either
- add produces and consumes in all these operations
- or add a global produces and consumes setting in this swagger file?
Feel free to take either approach you prefer

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

**Comment by ArcturusZhang** [comment]
Also the breakign change review required label is added because the breaking change pipeline runs into some issue and produces errors. This does not mean this PR contains breaking changes.
I am not sure if this failure is related with the consumes and produces issue - we could take a look again after that is fixed

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**